src/pybind/mgr/devicehealth/module.py

```diff
@@ -278,15 +279,78 @@ def config_notify(self) -> None:
                     self.get_module_option(opt['name']))
             self.log.debug(' %s = %s', opt['name'], getattr(self, opt['name']))
 
+    def _legacy_put_device_metrics(self, t: str, devid: str, data: str) -> None:
+        SQL = """
+        INSERT OR IGNORE INTO DeviceHealthMetrics (time, devid, raw_smart)
+            VALUES (?, ?, ?);
+        """
+
+        self._create_device(devid)
+        epoch = self._t2epoch(t)
+        json.loads(data)  # valid?
+        self.db.execute(SQL, (epoch, devid, data))
+
+    devre = r"[a-zA-Z0-9-]+[_-][a-zA-Z0-9-]+[_-][a-zA-Z0-9-]+"
+
+    def _load_legacy_object(self, ioctx: rados.Ioctx, oid: str) -> bool:
+        MAX_OMAP = 10000
+        self.log.debug(f"loading object {oid}")
+        if re.search(self.devre, oid) is None:
+            return False
+        with rados.ReadOpCtx() as op:
+            it, rc = ioctx.get_omap_vals(op, None, None, MAX_OMAP)
+            if rc == 0:
+                ioctx.operate_read_op(op, oid)
+                count = 0
+                for t, raw_smart in it:
+                    self.log.debug(f"putting {oid} {t}")
+                    self._legacy_put_device_metrics(t, oid, raw_smart)
+                    count += 1
+                assert count < MAX_OMAP
+        self.log.debug(f"removing object {oid}")
+        ioctx.remove_object(oid)
+        return True
+
+    def check_legacy_pool(self) -> bool:
+        try:
+            # 'device_health_metrics' is automatically renamed '.mgr' in
+            # create_mgr_pool
+            ioctx = self.rados.open_ioctx(self.MGR_POOL_NAME)
+        except rados.ObjectNotFound:
+            return True
+        if not ioctx:
+            return True
+
+        done = False
+        with ioctx:
+            with self._db_lock:
+                with self.db:
```

Comment:
nit, could use a more compacted form of `with` statement, like
```py
with ioctx, self._db_lock, self.db:
    count = 0
    # ...
```

see https://docs.python.org/3/reference/compound_stmts.html#the-with-statement. where multiple `with_item`s are allowed.

Reply:
What a relief, I'll change that.